src/main/java/org/jvnet/hudson/plugins/thinbackup/ThinBackupPluginImpl.java

```diff
@@ -60,20 +60,26 @@ public class ThinBackupPluginImpl extends Plugin {
     private boolean backupBuildsToKeepOnly = false;
     private boolean failFast = true;
 
-    @Override
-    public void start() throws Exception {
-        super.start();
+    public ThinBackupPluginImpl() {
+        // check if old config is there and no new config exists
+        final File oldConfig = new File(Jenkins.get().getRootDir(), "thinBackup.xml");
+        final File newConfig = getConfigFile().getFile();
+        boolean oldConfigExists = oldConfig.exists();
+        boolean newConfigExits = newConfig.exists();
+        if (oldConfigExists && !newConfigExits) {
+            LOGGER.warning("old config of 'thinBackup' detected, moving to new name");
+            try {
+                Files.move(oldConfig.toPath(), newConfig.toPath(), StandardCopyOption.REPLACE_EXISTING);
+            } catch (IOException e) {
+                LOGGER.severe("unable to move old config to new config, you will need to reconfigure thinBackup plugin manually");
+            }
+        }
```

Comment:
@basil I came up with a quite simple approach how to migrate the old to the new config. I will write a unit test for that for sure, but wanted to check if there is a "more common" approach in how to handle this

Reply:
This isn't very common, but in a few places we override `protected XmlFile getConfigFile()` to return a "legacy" XML file name rather than the standard fully qualified class name, for example in `ScriptlerConfiguration` (in Scriptler plugin) and `ProxyConfiguration` (in core). This approach could work here and save you from having to migrate the XML file to a new name. See jenkinsci/jenkins#8952 for an example of how to use `@LocalData` to test such a change.

Reply:
thanks that helped a lot!